app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddCyclewayPartSurface.kt

```diff
@@ -40,9 +41,18 @@ class AddCyclewayPartSurface : OsmFilterQuestType<SurfaceAnswer>() {
         answer.applyTo(tags, "cycleway")
         if (tags["cycleway:surface"] != null && tags["footway:surface"] != null) {
             if (tags["footway:surface"] == tags["cycleway:surface"]) {
-                tags["surface"] = tags["cycleway:surface"]!!
+                if (tags["surface"] != tags["cycleway:surface"]!!) {
+                    tags["surface"] = tags["cycleway:surface"]!!
+                    tags.remove("smoothness")
+                }
+            } else if (tags["footway:surface"] in ANYTHING_FULLY_PAVED && tags["cycleway:surface"] in ANYTHING_FULLY_PAVED) {
+                if (tags["surface"] != "paved") {
+                    tags["surface"] = "paved"
+                    tags.remove("smoothness")
+                }
```

Comment:
Well, I don't know about that...

1. As far as I remember, StreetComplete asks for the surface of all roads with `surface=paved` if not also a note is supplied. We do not really want that here, though

3. You seem to forget that there are categorizations below `paved` and `unpaved` possible. E.g. for `footway:surface=unhewn_cobblestone` , `cycleway:surface=sett` , `surface=cobblestone`, it would lead to `surface` being retagged to `paved`.

Reply:
> Well, I don't know about that...

I've decided to go that additional step, because there is [significant number of situations](https://overpass-turbo.eu/s/1nFA) (500 ways just for one city) where a combination of `asphalt` + `paving_stones` is popular for such segregated foot+bicycle ways, so removing `surface` _completely_ seems too harsh.   

> As far as I remember, StreetComplete asks for the surface of all roads with `surface=paved` if not also a note is supplied

- I can change `AddPathSurface.kt` to skip quest on `surface=paved` if both `footway:surface` and `cycleway:surface` are present (`AddRoadSurface.kt` and `AddPitchSurface.kt` which also check that are not relevant here)

- Alternatively,  I can add a `surface:note=different footway and cycleway surfaces` when I set `surface=paved` in this PR

Which one would you prefer?

> You seem to forget that there are categorizations below paved and unpaved possible

They are, but _**very few**_. For example, for `cobblestone`, there is [10 cases wordwide](https://overpass-turbo.eu/s/1nFt), so additional complexity did not seem worth it (also, all 10 cases currently existing are `footway:surface=sett` + `cycleway:surface=sett` + `surface=cobblestone`, which would correctly be converted by code to more precise `surface=sett`). But even if they were more popular, less precise `surface=paved` would be preferred to removal of `surface=*` (which is what currently happens), so this PR would be step in right direction.

Reply:
> so removing surface completely seems too harsh.

Well, it was a mistake before, just the mistake is removed. Also, StreetComplete will then just ask about the surface, doesn't it?

> I can change AddPathSurface.kt to skip quest on surface=paved if both footway:surface and cycleway:surface are present (AddRoadSurface.kt and AddPitchSurface.kt which also check that are not relevant here)
> Alternatively, I can add a surface:note=different footway and cycleway surfaces when I set surface=paved in this PR
> 
> Which one would you prefer?

Neither.

Reply:
> Well, it was a mistake before, just the mistake is removed.

I'm not sure if you talk about mistake in `v48` and older (i.e. invalid `surface=*` being left behind) or mistake in `v49.0` (i.e. `surface=*` being removed which is not wrong, but might have been handled better in some cases). If the latter, yes, this PR tries to **_improve on_** what the previous one (PR https://github.com/streetcomplete/StreetComplete/pull/4548) accomplished.

>  Also, StreetComplete will then just ask about the surface right after, doesn't it?

After some extra looking... No, it will not.

Which also means there is no need for workarounds I incorrectly suggested in last post. 
So, allow me to correct my answer on your initial remark:

>  As far as I remember, StreetComplete asks for the surface of all roads with surface=paved if not also a note is supplied

with "no, it will not ask for the `surface=*` on ways with `segregated=yes`, on which quests in this PR operate. "

Reply:
No, I talk about the mistake that would be introduced if the tagging was like this:

`footway:surface=paving_stones`
`surface=paving_stones`

and `cycleway:surface=asphalt` was added. 

And this is what you already implemented.

> No, not as far as I can tell, because path surface=* will be asked only on:

Ah okay, that's fine then.

I am a bit confused right now, I will revisit this PR later based on the information in this thread.

Reply:
OK. Previously, your example would remove `surface=*` tag. This PR improves that so in such example `surface=paved` would be added instead.

Reply:
This is also recommended way to tag it, see second block in `S5` on https://wiki.openstreetmap.org/wiki/Bicycle